### ISSUE TITLE
Error when creating new section (Undefined offset)

### ISSUE THREAD

**Issue by Michael-Reich** [spicywebau/craft-neo#189]
#### Description

When I work locally on MAMP Pro and I want to create a new Section I get this error:

```
PHP Notice – yii\base\ErrorException
Undefined offset: 2

yii\base\ErrorException: Undefined offset: 2 in /Users/michaelreich/SourceTree/arxonas-website/vendor/spicyweb/craft-neo/src/services/BlockTypes.php:354
Stack trace:
#0 /Users/michaelreich/SourceTree/arxonas-website/vendor/craftcms/cms/src/web/ErrorHandler.php(76): yii\base\ErrorHandler->handleError(8, 'Undefined offse...', '/Users/michaelr...', 354)
#1 /Users/michaelreich/SourceTree/arxonas-website/vendor/spicyweb/craft-neo/src/services/BlockTypes.php(354): craft\web\ErrorHandler->handleError(8, 'Undefined offse...', '/Users/michaelr...', 354, Array)
#2 /Users/michaelreich/SourceTree/arxonas-website/vendor/craftcms/cms/src/services/ProjectConfig.php(823): benf\neo\services\BlockTypes->handleChangedBlockType(Object(craft\events\ConfigEvent))
#3 [internal function]: craft\services\ProjectConfig->craft\services\{closure}(Object(craft\events\ConfigEvent))
#4 /Users/michaelreich/SourceTree/arxonas-website/vendor/yiisoft/yii2/base/Component.php(627): call_user_func(Object(Closure), Object(craft\events\ConfigEvent))
#5 /Users/michaelreich/SourceTree/arxonas-website/vendor/craftcms/cms/src/services/ProjectConfig.php(519): yii\base\Component->trigger('updateItem', Object(craft\events\ConfigEvent))
#6 /Users/michaelreich/SourceTree/arxonas-website/vendor/craftcms/cms/src/services/ProjectConfig.php(366): craft\services\ProjectConfig->processConfigChanges('neoBlockTypes.9...', true)
#7 /Users/michaelreich/SourceTree/arxonas-website/vendor/spicyweb/craft-neo/src/services/BlockTypes.php(229): craft\services\ProjectConfig->set('neoBlockTypes.9...', Array)
#8 /Users/michaelreich/SourceTree/arxonas-website/vendor/spicyweb/craft-neo/src/services/Fields.php(110): benf\neo\services\BlockTypes->save(Object(benf\neo\models\BlockType), false)
#9 /Users/michaelreich/SourceTree/arxonas-website/vendor/spicyweb/craft-neo/src/Field.php(550): benf\neo\services\Fields->save(Object(benf\neo\Field))
#10 /Users/michaelreich/SourceTree/arxonas-website/vendor/craftcms/cms/src/services/Fields.php(1537): benf\neo\Field->afterSave(false)
#11 /Users/michaelreich/SourceTree/arxonas-website/vendor/craftcms/cms/src/services/Fields.php(817): craft\services\Fields->applyFieldSave('87d558d5-38eb-4...', Array, 'global')
#12 /Users/michaelreich/SourceTree/arxonas-website/vendor/craftcms/cms/src/services/ProjectConfig.php(823): craft\services\Fields->handleChangedField(Object(craft\events\ConfigEvent))
#13 [internal function]: craft\services\ProjectConfig->craft\services\{closure}(Object(craft\events\ConfigEvent))
#14 /Users/michaelreich/SourceTree/arxonas-website/vendor/yiisoft/yii2/base/Component.php(627): call_user_func(Object(Closure), Object(craft\events\ConfigEvent))
#15 /Users/michaelreich/SourceTree/arxonas-website/vendor/craftcms/cms/src/services/ProjectConfig.php(519): yii\base\Component->trigger('updateItem', Object(craft\events\ConfigEvent))
#16 /Users/michaelreich/SourceTree/arxonas-website/vendor/craftcms/cms/src/helpers/ProjectConfig.php(63): craft\services\ProjectConfig->processConfigChanges('fields.87d558d5...')
#17 /Users/michaelreich/SourceTree/arxonas-website/vendor/craftcms/cms/src/services/Sections.php(571): craft\helpers\ProjectConfig::ensureAllFieldsProcessed()
#18 /Users/michaelreich/SourceTree/arxonas-website/vendor/craftcms/cms/src/services/ProjectConfig.php(823): craft\services\Sections->handleChangedSection(Object(craft\events\ConfigEvent))
#19 [internal function]: craft\services\ProjectConfig->craft\services\{closure}(Object(craft\events\ConfigEvent))
#20 /Users/michaelreich/SourceTree/arxonas-website/vendor/yiisoft/yii2/base/Component.php(627): call_user_func(Object(Closure), Object(craft\events\ConfigEvent))
#21 /Users/michaelreich/SourceTree/arxonas-website/vendor/craftcms/cms/src/services/ProjectConfig.php(512): yii\base\Component->trigger('addItem', Object(craft\events\ConfigEvent))
#22 /Users/michaelreich/SourceTree/arxonas-website/vendor/craftcms/cms/src/services/ProjectConfig.php(366): craft\services\ProjectConfig->processConfigChanges('sections.a0e75d...', true)
#23 /Users/michaelreich/SourceTree/arxonas-website/vendor/craftcms/cms/src/services/Sections.php(487): craft\services\ProjectConfig->set('sections.a0e75d...', Array)
#24 /Users/michaelreich/SourceTree/arxonas-website/vendor/craftcms/cms/src/controllers/SectionsController.php(186): craft\services\Sections->saveSection(Object(craft\models\Section))
#25 [internal function]: craft\controllers\SectionsController->actionSaveSection()
#26 /Users/michaelreich/SourceTree/arxonas-website/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#27 /Users/michaelreich/SourceTree/arxonas-website/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#28 /Users/michaelreich/SourceTree/arxonas-website/vendor/craftcms/cms/src/web/Controller.php(109): yii\base\Controller->runAction('save-section', Array)
#29 /Users/michaelreich/SourceTree/arxonas-website/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('save-section', Array)
#30 /Users/michaelreich/SourceTree/arxonas-website/vendor/craftcms/cms/src/web/Application.php(297): yii\base\Module->runAction('sections/save-s...', Array)
#31 /Users/michaelreich/SourceTree/arxonas-website/vendor/craftcms/cms/src/web/Application.php(561): craft\web\Application->runAction('sections/save-s...', Array)
#32 /Users/michaelreich/SourceTree/arxonas-website/vendor/craftcms/cms/src/web/Application.php(281): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#33 /Users/michaelreich/SourceTree/arxonas-website/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#34 /Users/michaelreich/SourceTree/arxonas-website/web/index.php(21): yii\base\Application->run()
#35 {main}
```

#### Steps to reproduce
1.
2.
3.

#### Other information
- Neo version: Latest
- Craft version: Latest
- Is the affected Craft installation multi-site:
- (if multi-site) Are the affected Neo fields set to manage blocks on a per-site basis:

**Comment by ttempleton** [comment]
Thanks for reporting, @Michael-Reich.  This has now been fixed in Neo 2.2.4.

**Comment by Michael-Reich** [comment]
That's fast support, thank you very much!
